src/client/activation/types.ts

```diff
@@ -14,6 +14,10 @@ export const IExtensionActivationManager = Symbol('IExtensionActivationManager')
  * @extends {IDisposable}
  */
 export interface IExtensionActivationManager extends IDisposable {
+    /**
+     * Method invoked when extension activates (invoked once).
+     */
+    activateSync(): void;
```

Comment:
Found having a sync version is better, sometimes we need things to happen before other things.